RDKShell/RDKShell.cpp

```diff
@@ -5722,6 +5729,67 @@ namespace WPEFramework {
             returnResponse(status);
         }
 
+        uint32_t RDKShell::checkpointWrapper(const JsonObject& parameters, JsonObject& response)
+        {
+            LOGINFOMETHOD();
+            bool status = false;
+            if (parameters.HasLabel("callsign"))
+            {
+                std::string callsign = parameters["callsign"].String();
+                bool isApplicationBeingDestroyed = false;
+
+                gLaunchDestroyMutex.lock();
+                if (gDestroyApplications.find(callsign) != gDestroyApplications.end())
+                {
+                    isApplicationBeingDestroyed = true;
+                }
+                if (gExternalDestroyApplications.find(callsign) != gExternalDestroyApplications.end())
+                {
+                    isApplicationBeingDestroyed = true;
+                }
+                gLaunchDestroyMutex.unlock();
+
+                if (isApplicationBeingDestroyed)
+                {
+                    std::cout << "ignoring checkpoint for " << callsign << " as it is being destroyed " << std::endl;
+                    status = false;
+                    response["message"] = "failed to checkpoint application, is being destroye";
+                    returnResponse(status);
+                }
+
+                WPEFramework::Core::JSON::String stateString;
+                const string callsignWithVersion = callsign + ".1";
+
+                uint32_t stateStatus = getThunderControllerClient(callsignWithVersion)->Get<WPEFramework::Core::JSON::String>(RDKSHELL_THUNDER_TIMEOUT, "state", stateString);
+
+                if (stateStatus == 0 && stateString != "suspended")
+                {
+                    std::cout << "ignoring checkpoint for " << callsign << " as it is not suspended " << std::endl;
+                    status = false;
+                    response["message"] = "failed to checkpoint application, not suspended";
+                    returnResponse(status);
+                }
+
+                status = mMemCheckpointRestoreClient.checkpoint(callsign);
+            }
+
+            returnResponse(status);
+        }
+
+        uint32_t RDKShell::restoreWrapper(const JsonObject& parameters, JsonObject& response)
+        {
+            LOGINFOMETHOD();
+            bool status = false;
+            if (parameters.HasLabel("callsign"))
+            {
+                std::string callsign = parameters["callsign"].String();
```

Comment:
Shouldn't we check if it is checkpointed based on ie internal mMemCheckpointRestoreClient list with checkpointed pids/callSigns ?